src/main/extension/extension_install.cpp

```diff
@@ -244,20 +244,46 @@ void ExtensionHelper::InstallExtensionInternal(DBConfig &config, ClientConfig *c
 		throw IOException("No slash in URL template");
 	}
 
-	// Special case to install extension from a local file, useful for testing
+	// Special case to install extension from a local file or NOT plain http
+	// Note that before making this the default path, the following are needed for feature-parity:
+	//	* On success (name.duckdb_extension.gz present): Performing a single GET (instead of HEAD + potentially multiple
+	//GETs)
+	//	* On failure (name.duckdb_extension[.gz] not presents): Have a way to fail with a single GET (instead of
+	//currently 2 HEADs)
+	//	* Setting the User-Agent headers
 	if (!StringUtil::Contains(url_template, "http://")) {
 		string file = fs.ConvertSeparators(url);
+		bool endsInGz = true;
 		if (!fs.FileExists(file)) {
-			// check for non-gzipped variant
-			file = file.substr(0, file.size() - 3);
-			if (!fs.FileExists(file)) {
-				throw IOException("Failed to copy local extension \"%s\" at PATH \"%s\"\n", extension_name, file);
+			string message_verb = FileSystem::IsRemoteFile(file) ? "download" : "copy local";
+			string addendum = FileSystem::IsRemoteFile(file) ? "\n'LOAD httpfs' might be required." : "";
+			if (StringUtil::EndsWith(file, ".gz")) {
+				// check also for non-gzipped variant
+				file = file.substr(0, file.size() - 3);
+				if (!fs.FileExists(file)) {
+					throw IOException("Failed to %s extension \"%s\" at either of:\n     \"%s.gz\" (gzip "
+					                  "compressed version) or\n     \"%s\" (no encoding)\n%s",
+					                  message_verb, extension_name, file, file, addendum);
+				}
+				endsInGz = false;
```

Comment:
I think this code is slightly obfuscated, removing endsInGz en rechecking the StringUtil::EndsWith(file, ".gz") in the if statement that now checks endsInGz would make things easier to read.

Reply:
Thanks, improved.